crates/oauth2-types/src/requests.rs

```diff
@@ -91,90 +135,226 @@ pub enum Display {
+    /// Hint to the Authorization Server about the login identifier the End-User
+    /// might use to log in.
+    pub login_hint: Option<String>,
 
+    /// Requested Authentication Context Class Reference values.
     #[serde_as(as = "Option<StringWithSeparator::<SpaceSeparator, String>>")]
     #[serde(default)]
-    acr_values: Option<HashSet<String>>,
+    pub acr_values: Option<HashSet<String>>,
 
+    /// A JWT that contains the request's parameter values, called a [Request
+    /// Object].
+    ///
+    /// [Request Object]: https://openid.net/specs/openid-connect-core-1_0.html#RequestObject
     pub request: Option<String>,
 
+    /// A URI referencing a [Request Object] or a [Pushed Authorization
+    /// Request].
+    ///
+    /// [Request Object]: https://openid.net/specs/openid-connect-core-1_0.html#RequestUriParameter
+    /// [Pushed Authorization Request]: https://datatracker.ietf.org/doc/html/rfc9126
     pub request_uri: Option<Url>,
 
+    /// A JSON object containing the Client Metadata when interacting with a
+    /// [Self-Issued OpenID Provider].
+    ///
+    /// [Self-Issued OpenID Provider]: https://openid.net/specs/openid-connect-core-1_0.html#SelfIssued
     pub registration: Option<String>,
 }
 
-#[derive(Serialize, Deserialize, Default)]
+impl AuthorizationRequest {
+    /// Creates a basic `AuthorizationRequest`.
+    #[must_use]
+    pub fn new(
+        response_type: OAuthAuthorizationEndpointResponseType,
+        client_id: String,
+        scope: Scope,
+    ) -> Self {
+        Self {
+            response_type,
+            client_id,
+            redirect_uri: None,
+            scope,
+            state: None,
+            response_mode: None,
+            nonce: None,
+            display: None,
+            prompt: None,
+            max_age: None,
+            ui_locales: None,
+            id_token_hint: None,
+            login_hint: None,
+            acr_values: None,
+            request: None,
+            request_uri: None,
+            registration: None,
+        }
+    }
+}
+
+/// A successful response from the [Authorization Endpoint].
+///
+/// [Authorization Endpoint]: https://www.rfc-editor.org/rfc/rfc6749.html#section-3.1
+#[skip_serializing_none]
+#[derive(Serialize, Deserialize, Default, Debug, Clone)]
 pub struct AuthorizationResponse<R> {
     pub code: Option<String>,
     #[serde(flatten)]
     pub response: R,
 }
 
+/// A request to the [Token Endpoint] for the [Authorization Code] grant type.
+///
+/// [Token Endpoint]: https://www.rfc-editor.org/rfc/rfc6749#section-3.2
+/// [Authorization Code]: https://www.rfc-editor.org/rfc/rfc6749#section-4.1
 #[skip_serializing_none]
-#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 pub struct AuthorizationCodeGrant {
+    /// The authorization code that was returned from the authorization
+    /// endpoint.
     pub code: String,
-    #[serde(default)]
```

Comment:
Haven't tested, but there are multiple places where you deleted `#[serde(default)]` annotations. Isn't that a problem when deserialising, and the field is absent?

Reply:
The attribute is not needed on `Option`s. It defaults to `None` automatically.